src/plugins/intel_cpu/src/nodes/concat.cpp

```diff
@@ -364,37 +369,62 @@ void Concat::prepareParams() {
     if (getSelectedPrimitiveDescriptor() == nullptr)
         IE_THROW() << "Preferable primitive descriptor is not set.";
 
-    std::vector<memory::desc> srcs_d;
-    for (size_t i = 0; i < getParentEdges().size(); i++) {
-        const auto& srcMemPtr = getParentEdgesAtPort(i)[0]->getMemoryPtr();
-        if (!srcMemPtr || !srcMemPtr->isAllocated()) {
-            auto parent = getParentEdgeAt(i)->getParent();
-            IE_THROW() << "Source memory from " << parent->getName() << " didn't allocate for node "
-                               << getName() << ".";
+        auto outputStride = dstMemPtr->getDescPtr()->as<BlockedMemoryDesc>()->getStrides();
+        size_t curConcatOffset = 0;
+        const size_t elemSize = DnnlExtensionUtils::sizeOfDataType(dstMemPtr->GetDataType());
+        const auto& outputOrder = getParentEdgesAtPort(0)[0]->getMemoryPtr()->getDescPtr()->as<BlockedMemoryDesc>()->getOrder();
+        for (size_t i = 0; i < outputOrder.size(); i++) {
+            if (outputOrder[i] == axis) {
+                reorderedAxis = i;
+                break;
+            }
         }
+        std::vector<memory::desc> srcs_d;
+        for (size_t i = 0; i < getParentEdges().size(); i++) {
+            const auto& srcMemPtr = getParentEdgesAtPort(i)[0]->getMemoryPtr();
+            if (!srcMemPtr || !srcMemPtr->isAllocated()) {
+                auto parent = getParentEdgeAt(i)->getParent();
+                IE_THROW() << "Source memory from " << parent->getName() << " didn't allocate for node "
+                           << getName() << ".";
+            }
 
-        if (srcMemPtr->GetShape().hasZeroDims()) {
-            continue;
-        }
+        if (canExecRef) {
+            const auto& inputShape = srcMemPtr->getDescPtr()->as<BlockedMemoryDesc>()->getBlockDims();
+            inputStrides[i] = srcMemPtr->getDescPtr()->as<BlockedMemoryDesc>()->getStrides();
+            size_t nElem = 1;
+            for (size_t j = reorderedAxis; j < inputShape.size(); j++) {
+                nElem *= inputShape[j];
+            }
+            nelemToCopy[i] = nElem * elemSize;
+            dstOffset[i] = outputStride[reorderedAxis] * curConcatOffset * elemSize;
+            curConcatOffset += inputShape[reorderedAxis];
+            auto desc = srcMemPtr->GetDescWithType<DnnlMemoryDesc>()->getDnnlDesc();
+            srcsOffset[i] = desc.data.offset0;
```

Comment:
```suggestion
            srcsOffset[i] = srcMemDesc->getOffsetPadding()
```
Helps to avoid possible desc conversions and downcast.